keymanagementrequirements.md

```diff
@@ -0,0 +1,79 @@
+- Signature Validity (Expiry, Revocation and/or Trusted Update List. Mechanism to update information on the validity of signatures.)
+
+## Personas:
+- Publisher: User who builds and signs containers.
+    - Publisher Admin: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Admin users (i.e. security administrator) will be responsible for configuring roots, provide access to use or generate delegate keys, and make decisions for key revocation.
+    - Publisher Builder: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Builders (i.e. developers, build hosts) will be responsible for creating artifacts. 
+    - Publisher Signer: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Signers (i.e. developers, signing hosts) will be responsible for signing artifacts.
+- Deployer: User who deploys containers.
+    - Deployer Admin: In some scenarios a deployer will include a group of users (i.e. teams or enterprises). Admin users (i.e. security administrators) will be responsible for configuring deployment policies, including potentially a set of trusted roots or required signatures. 
+    - Deployer Operator: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Operators (i.e. developers, deployment hosts) will be responsible for pulling container images from registries, verifying their signatures, and then running them.
+- Repository Owner: In some scenarios a container may be stored in a repository that is managed by the repository owner. The repository may be public or private and could also be air-gapped.
+
+## Definitions:
+- Root key: A self signed key used for the lowest designation of trust. Root keys can be created by developers, organizations, public/private CAs, and registry operators. The root key should be retrieved from a trusted source that can establish the authenticity of the creator's identity.
+- Signing key: A signing key is used to generate artifact signatures. A signing key should be signed with a root key or one of its intermediaries. The certificate chain with a signing key can be used to verify which root it belongs to. While a root key can be used as a signing key, this is not recommended as it creates a large blast radius and increases the risk of compromising a root key. 
+- Trust Policy: The trust policy defines whether to enable signature validation and which checks need to be run. An example trust policy:
+```
+    {
+        "trustPolicy": {
+            "signatureCheck": true,
+            "optionalChecks": {
+            	"signatureExpiry": true,
+            	"signatureRescinded": false
+            },
+            "trustStore": [
+            	"truststore.json",
+            	{
+            		"trustedRoots": [{
+            			"root": "test.crt"
+            		}]
+            	}
```

Comment:
It is a bit strange to have both a string and a JSON object in an JSON array although it is valid.

Reply:
This was meant to be an illustrative example to make the requirements clearer. It's allowing a reference to a local file as well as specifying the roots directly in the policy document. We can come up with a clean implementation when defining the specifications for the documents.